sunpy/net/dataretriever/client.py

```diff
@@ -336,6 +337,15 @@ def fetch(self, qres, path=None, error_callback=None, **kwargs):
         -------
         Results Object
         """
+        # Check for type of path
+        if isinstance(path, str) or path is None:
+            pass
+        elif isinstance(path, pathlib.Path):
+            path = str(path.absolute())
+        else:
+            err = 'path should be either pathlib.Path object or str object. '\
```

Comment:
Not sure I like using the word object here.

Reply:
What is a 'word' object?